lib/rex/post/meterpreter/ui/console/command_dispatcher/stdapi/ui.rb

```diff
@@ -168,6 +170,166 @@ def cmd_screenshot( *args )
+      "-d" => [ true, "The stream duration in seconds (Default: 1800)" ] # 30 min
+    )
+
+    screenshare_opts.parse( args ) { | opt, idx, val |
+      case opt
+        when "-h"
+          print_line( "Usage: screenshare [options]\n" )
+          print_line( "View the current interactive desktop in real time." )
+          print_line( screenshare_opts.usage )
+          return
+        when "-q"
+          quality = val.to_i
+        when "-p"
+          path = val
+        when "-v"
+          view = true if ( val =~ /^(t|y|1)/i )
+        when "-d"
+          duration = val.to_i
+      end
+    }
+
+    print_status("Preparing player...")
+
+    html = %|<html>
+<head>
+<META HTTP-EQUIV="PRAGMA" CONTENT="NO-CACHE, NO-STORE">
+<META HTTP-EQUIV="CACHE-CONTROL" CONTENT="NO-CACHE, NO-STORE">
+<title>Metasploit screenshare stream - #{client.sock.peerhost}</title>
+<script language="javascript">
+function updateStatus(msg) {
+  var status = document.getElementById("status");
+  status.innerText = msg;
+}
+
+
+function noImage() {
+  document.getElementById("streamer").style = "display:none";
+  updateStatus("Waiting");
+}
+
+var i = 0;
+var imgurl;
+var imgElement;
+var didload = 1;
+function updateFrame() {
+  didload = 0;
+  imgurl = "#{stream_path}?" + i;
+  getImage(imgurl, i).then(
+    function(e){
+        didload = 1;
+        newImg = e.img;
+        index = e.index;
+        var theWrap = document.getElementById("vid_cont");
+        ni = theWrap.appendChild(newImg);
+        //console.log('added img_'+index);
+        ni.id = "img_"+index;
+        purge = document.querySelector("img:not(#img_"+index+")")
+        if (typeof(purge) != 'undefined' && purge != null){
+          purge.remove();
+        }
+        updateStatus("Playing");
+        
+    },
+    function(errorurl, index){
+        //console.log('Error loading ' + errorurl)
+        updateStatus("Waiting");
+        didload = 1;
+    }
+  );
+}
+
+function getImage(url, index){
+    return new Promise(function(resolve, reject){
+        img = new Image();
+        img.onload = function(){
+            var e = new Object();
+            e.img = img;
+            e.index = index;
+            resolve(e);
+        }
+        img.onerror = function(){
+            reject(index);
+        }
+        img.src = url;
+    })
+}
+
+setInterval(function() {
+  if(didload){
+   updateFrame();
+   i++; 
+  }else{
+    //console.log('skipping');
+  }
+},42);
+
+</script>
+</head>
+<body>
+<noscript>
+  <h2><font color="red">Error: You need Javascript enabled to watch the stream.</font></h2>
+</noscript>
+<pre>
+Target IP  : #{client.sock.peerhost}
+Start time : #{Time.now}
+Status     : <span id="status"></span>
+</pre>
+<br>
+<div id="vid_cont">
+  
+</div>
+<br><br>
+</body>
+</html>
+    |
+
+    ::File.open(player_path, 'wb') do |f|
+      f.write(html)
+    end
+
+    print_status("Please open the player manually with a browser: #{player_path}")
```

Comment:
what about `if view`?

Reply:
The method should create the HTML player regardless of whether view is true or not.  My understanding of the -v flag was that it was a conditional to auto-launch the viewer or not.  Without the HTML file generates, there really is no way to view the stream (webcam or desktop) and the method becomes useless in it's current configuration